wp-includes/service-workers.php

```diff
@@ -334,3 +334,25 @@ function wp_service_worker_fragment_redirect_old_slug_to_new_url( $link ) {
 	}
 	return $link;
 }
+
+/**
+ * Get service worker error messages.
+ *
+ * @return array Array of error messages: default, comment.
+ */
+function wp_service_worker_get_error_messages() {
+	return apply_filters(
+		'wp_service_worker_error_messages',
+		array(
+			'default' => __( 'Please check your internet connection, and try again.', 'pwa' ),
+			'comment' => __( 'Your comment will be submitted once you are back online!', 'pwa' ),
+		)
+	);
+}
+
+/**
+ * Display service worker error message template tag.
+ */
+function wp_service_worker_error_message_placeholder() {
+	echo wp_kses_post( '<p><!--WP_SERVICE_WORKER_ERROR_MESSAGE--></p>' );
```

Comment:
I don't think the `wp_kses_post()` is necessary here. Also, should the paragraphs be included?

Reply:
Not sure about the paragraphs -- I was thinking at first that maybe not but then the developer would always have to add the markup instead of just overriding the text message. Without markup it would look like this:
![screen shot 2018-10-23 at 11 15 47 am](https://user-images.githubusercontent.com/3294597/47366957-1a17eb80-d6b5-11e8-870c-a29b9f41cffc.png)

So basically I was thinking that maybe for the default template we should add paragraph and try to avoid "breaking" how the default offline template looks like.

However, I'm fine removing it.

Reply:
OK, for now let's just do:

```php
echo '<p><!--WP_SERVICE_WORKER_ERROR_MESSAGE--></p>' ;
```

(No need for `wp_kses_post()`.)

I think I was thinking of a case where there is no message. But I suppose there would always be one.